app/models/vm_or_template.rb

```diff
@@ -393,7 +393,7 @@ def self.task_arguments(options)
     when "remove_snapshot", "revert_to_snapshot" then
       [options[:snap_selected]]
     when "create_snapshot" then
-      [options[:name], options[:description], options[:memory]]
+      [options[:name], options[:description], options.fetch(:memory, false)]
```

Comment:
This approach fixes the original issue and adds support for passing the memory flag from automate.   Also we want to move the snapshot methods out of the base automate service model and into the VMware sub-classed model where it belongs since these methods are currently not supported on other providers.